### PR TITLE
feat: add scaffold example specs UI

### PR DESCRIPTION
**Pull request by ZachJW34**
- Closes [UNIFY-629](https://cypress-io.atlassian.net/jira/software/projects/UNIFY/boards/20?selectedIssue=UNIFY-629)

### User facing changelog
Adds UI for scaffolding example integration specs

### Additional details
Adds tree view of generated example specs to spec-generator modals. Also tweaks the spec generation such that closing the modal resets the state of the flow.

### How has the user experience changed?

https://user-images.githubusercontent.com/25158820/144317140-15d76a44-b859-4085-85c8-5241b64c85a3.mov

### How To Test
Open any Cypress project (choosing e2e), then click on the `NewSpec` button in the top right of the home page and click `Scaffold Example Specs` in the modal or (if you have no specs) click on the `Scaffold Example Specs` tile that is on the home page.

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/app/src/specs/CreateSpecModal.vue [hunk 1]
```diff
@@ -6,7 +6,7 @@
     :title="title"
     :model-value="show"
     data-testid="create-spec-modal"
-    @update:model-value="$emit('close')"
+    @update:model-value="close"
```
Comment (ImCesar) [inline packages/app/src/specs/CreateSpecModal.vue:9]: The standard modal has a help link by default and it seems to take you to the homepage of the docs. Is there anyway to disable that help link or send it somewhere useful?

**Comment by ImCesar** [comment]
Tested it with no e2e specs and with some and both cases worked fine.

File: packages/app/src/specs/CreateSpecModal.vue (hunk 1 above)
Comment (ZachJW34) [reply packages/app/src/specs/CreateSpecModal.vue:9]: I believe its supposed to be a general "catch-all" link, not sure about disabling but as it stands all generators have that for now so I won't make a change in this PR.

Comment (JessicaSachs) [reply packages/app/src/specs/CreateSpecModal.vue:9]: @ImCesar It's a huge pain to control outside of the modal. That's what `:title` is for. We can easily add a link to come back from the child.

**Review by JessicaSachs — APPROVED** [review]

**Pull request merged**